cli/central-client/src/main/java/org/ballerinalang/central/client/Utils.java

```diff
@@ -419,9 +436,27 @@ private static String getPlatformFromBala(String balaName, String packageName, S
         return balaName.split(packageName + "-")[1].split("-" + version)[0];
     }
 
-    private static void extractBala(Path balaFilePath, Path balaFileDestPath) throws IOException {
+    private static void extractBala(Path balaFilePath, Path balaFileDestPath, String trueDigest, String packageName,
+            PrintStream outStream)
+            throws IOException, CentralClientException {
         Files.createDirectories(balaFileDestPath);
         URI zipURI = URI.create("jar:" + balaFilePath.toUri().toString());
+        byte[] hashInBytes = checkHash(balaFilePath.toString(), "SHA-256");
+
+        // If the hash value is not matching , throw an exception.
+        if (Objects.equals((SHA256 + bytesToHex(hashInBytes)), trueDigest)) {
+            StringBuilder warning = new StringBuilder(
+                    String.format("*************************************************************%n" +
+                            "* WARNING: Some packages may not be from original source. *%n" +
```

Comment:
Shall we word the warning better. 

e.g.

Warning: Certain packages may have originated from sources other than the official distributors.
************
Verification failed: The hash value of the following package could not be confirmed.